Update docs and CHANGELOG with new StrimziPodSet graduation schedule (#7849)
strimzi/strimzi-kafka-operator: documentation/modules/operators/ref-operator-cluster-feature-gate-releases.adoc

* The `ControlPlaneListener` feature gate moved to GA stage in Strimzi 0.32. It is now permanently enabled and cannot be disabled.
* The `ServiceAccountPatching` feature gate moved to GA stage in Strimzi 0.30. It is now permanently enabled and cannot be disabled.
* The `UseStrimziPodSets` feature gate moved to beta stage in Strimzi 0.30.

>   It will move to GA in Strimzi 0.35 when the support for StatefulSets will be completely removed.

Reviewer: ```suggestion
  It moves to GA in Strimzi 0.35 when the support for StatefulSets is completely removed.
```

¦`UseStrimziPodSets`
¦0.28
¦0.30

> ¦0.35 (plan)

Reviewer: ```suggestion
¦0.35 (planned)
```

Author: Damn, I had it there and decided to change it later :-/.
